### ISSUE TITLE
[X86] Function multiversioning doesn't consistently handle features with priority=0

### ISSUE THREAD

**Issue by DanShaders** [llvm/llvm-project#98244]
The following code produces a correctly constructed ifunc resolver:
```cpp
[[gnu::target("sha")]] int f() { return 2; }
[[gnu::target("default")]] int f() { return 1; }

int main() { return f(); }
```

While in the snippet below, Clang either silently ignores the "sha" overload without assertions or ICEs with assertions.
```cpp
[[gnu::target("default")]] int f() { return 1; }
[[gnu::target("sha")]] int f() { return 2; }

int main() { return f(); }
```

CE link: https://godbolt.org/z/Ghzb6KzGf

The buggy behavior arises because "sha" has PRIORITY=0 as defined here:
https://github.com/llvm/llvm-project/blob/7bfb98c34687d9784f36937c3ff3e735698b498a/llvm/include/llvm/TargetParser/X86TargetParser.def#L214

Obviously, this is one of the manifestations of a long-standing issue with `__attribute__((target))` and "unorderable" features, so I'm not sure what the expected behavior is here. However, at the very least, issuing a diagnostic in the cases above will greatly improve DX.

**Comment by phoebewang** [comment]
CC @FreddyLeaf

**Comment by FreddyLeaf** [comment]
Yes, this is a known issue. gcc also can't order these features with priority=0 here. 
>However, at the very least, issuing a diagnostic in the cases above will greatly improve DX.

I agree.

**Comment by DanShaders** [comment]
I think it would be very cool if clang could understand something like:
```cpp
[[clang::target("default", 0)]] int f() { ... }
[[clang::target("avx2", 1)]] int f() { ... }
[[clang::target("sha", 2)]] int f() { ... }
```
and emit an error if explicit priority was not given.

Note that GCC devs may actually like this too based on [[1]](https://gcc.gnu.org/pipermail/gcc-patches/2012-June/341270.html#:~:text=This%20is%20not%20a%20perfect%20solution%20and%20in%20future%2C%20the%20user%0Ashould%20be%20allowed%20to%20assign%20a%20dispatching%20priority%20value%20to%20each%20version.) and [[2]](https://gcc.gnu.org/pipermail/gcc/2022-July/239148.html#:~:text=Note%20we%20face%20the%20very%20same%20problem%20on%20i386%2C%20where%20it%27s%20very%20hard%20to%20specify%20a%20priority%0A%20%20%20for%20the%20family%20of%20avx512%20features.%20That%20said%2C%20an%20optional%20priority%20specifier%20might%20be%20possible.).
